cassandane/Cassandane/Cyrus/TestCase.pm

```diff
@@ -1179,6 +1181,7 @@ sub run_replication
     push(@cmd, '-O') if defined $nosyncback;
     push(@cmd, '-u', $user) if defined $user;
     push(@cmd, '-m', $mailbox) if defined $mailbox;
+    push(@cmd, '-A') if defined $allusers;
```

Comment:
I think this `defined` is nqr... if you `run_replication(allusers => 0);` then you'll unexpectedly get the -A behaviour, even though you though you explicitly said you didn't want it, because `0` is not `undef`.

`defined` is kinda bogus for the existing options too, but it's less likely to cause a real problem because `run_replication(user => 0)`, `run_replication(mailbox => '')`, etc aren't really coherent things to try to do anyway.

```suggestion
    push(@cmd, '-A') if $allusers;  # n.b. boolean
```

... or fix them all, so that $allusers doesn't need a comment about why it's different from the `defined` ones